### PR TITLE
Stabilize allow_fail flag test feature

### LINKED ISSUE #46488: Stabilize allow_fail test flag
Implemented in #42219.

I'll make the stabilization.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #46488.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Rebased.

**Comment by nrc** [comment]
I think this needs some discussion before stabilisation - there was quite a lot of discussion on the original PR about whether this was a good idea or not, and I don't see any indication that this feature has proved itself in the meantime. Let's discuss at the next dev-tools meeting.

**Comment by kennytm** [comment]
Triage ping @rust-lang/dev-tools! Has this been discussed and is there a conclusion on this PR?

File: src/libsyntax/feature_gate.rs [hunk 1]
```diff
@@ -519,6 +516,8 @@ declare_features! (
     // Allows the sysV64 ABI to be specified on all platforms
     // instead of just the platforms on which it is the C ABI
     (accepted, abi_sysv64, "1.24.0", Some(36167)),
+    // Allows a test to fail without failing the whole suite
+    (accepted, allow_fail, "1.23.0", Some(42219)),
```
Comment (killercup) [inline src/libsyntax/feature_gate.rs]: just a quick note: this should be ≥1.25.0 by now

Comment (GuillaumeGomez) [reply src/libsyntax/feature_gate.rs]: Oh indeed!

**Comment by GuillaumeGomez** [comment]
@kennytm: The meeting is tomorrow so I hope we'll discuss it at this time.

**Comment by nrc** [comment]
Discussed at dev-tools meeting, we plan not to stabilise this feature

**Pull request closed without merging**